pkg/controller/operator/common/util.go

```diff
@@ -74,22 +74,27 @@ var (
 )
 
 func isKubermaticConfiguration(ref metav1.OwnerReference) bool {
-	return ref.APIVersion == operatorv1alpha1.SchemeGroupVersion.String() && ref.Kind == "KubermaticConfiguration"
+	return ref.APIVersion == kubermaticv1.SchemeGroupVersion.String() && ref.Kind == "KubermaticConfiguration"
 }
 
 func isSeed(ref metav1.OwnerReference) bool {
 	return ref.APIVersion == kubermaticv1.SchemeGroupVersion.String() && ref.Kind == "Seed"
 }
 
 // StringifyFeatureGates takes a set of enabled features and returns a comma-separated
-// key=value list like "featureA=true,featureB=true,...".
-func StringifyFeatureGates(cfg *operatorv1alpha1.KubermaticConfiguration) string {
-	features := make([]string, 0)
-	for _, feature := range cfg.Spec.FeatureGates.List() {
-		features = append(features, fmt.Sprintf("%s=true", feature))
+// key=value list like "featureA=true,featureB=true,...". The list of feature gates is
+// sorted, so the output of this function is stable.
+func StringifyFeatureGates(cfg *kubermaticv1.KubermaticConfiguration) string {
+	// use a set to ensure that the result is sorted, otherwise reconciling code that
+	// uses this will end up in endless loops
+	features := sets.NewString()
+	for feature, enabled := range cfg.Spec.FeatureGates {
+		if enabled {
```

Comment:
I know it wasn't in the original function, but what if some feature gate is being explicitly disabled? Looks like we simply skip (and destroy) this information.